### PR TITLE
Fluid form

### PR DESCRIPTION
**Pull request by aagonzales**
Closes https://github.com/carbon-design-system/carbon/issues/12453
Closes https://github.com/carbon-design-system/carbon-website/issues/1907
Closes https://github.com/carbon-design-system/carbon-website/issues/3328

Cleaned up form guidance and added documentation for fluid form.

#### Testing

Go to Components/Form and proof read the Usage and Style tab.

### CODE REVIEW

File: src/pages/components/form/usage.mdx [hunk 1]
```diff
@@ -117,215 +115,378 @@ do you gain by collecting this information?
...
 
-![effective form design](images/form-usage-1.png)
+<Tabs>
 
-</Column>
-</Row>
+<Tab label="Fixed">
 
-### Button placement
+![Anatomy of a fixed form](images/form-usage-1.png)
```
Comment (laurenmrice) [inline src/pages/components/form/usage.mdx:127]: Anatomy image (fixed):
This abbreviation should be `etc.` not ect.
Screen Shot 2023-01-24 at 12 44 54 PM

File: src/pages/components/form/usage.mdx [hunk 2]
```diff
@@ -117,215 +115,378 @@ do you gain by collecting this information?
...
-On non-modal or in-page forms, align single buttons or Secondary / Primary
-button groups to the left. Buttons should align with the form controls
-regardless of the user‘s window width.
+</Tab>
 
-<Row>
-<Column colLg={8}>
+<Tab label="Fluid">
+
+![Anatomy of a fluid form](images/form-usage-1-fluid.png)
```
Comment (laurenmrice) [inline src/pages/components/form/usage.mdx:133]: Anatomy image (fluid):
The spacing between the text and the icon should be 8px, but I think it's at 16px? 
Screen Shot 2023-01-24 at 1 05 41 PM

File: src/pages/components/form/usage.mdx [hunk 3]
```diff
@@ -117,215 +115,378 @@ do you gain by collecting this information?
...
+
+- Use sentence-style capitalization, and in most cases, write the text as full
+  sentences with punctuation.
+- When used, helper text that appears persistently underneath the field is
+  replaced by an error or warning message if a state change occurs.
+
+### Tooltip help
+
+Tooltips can be added inline with a form label if additional non-pertinent
+information is need to assist the user in correctly completing a field, such as
```
Comment (laurenmrice) [inline src/pages/components/form/usage.mdx:285]:
```suggestion
information is needed to assist the user in correctly completing a field, such as
```

File: src/pages/components/form/usage.mdx [hunk 4]
```diff
@@ -117,215 +115,378 @@ do you gain by collecting this information?
...
+context setting. Tooltip content may include more in-depth information about the
+field or the options the user is choosing from. Links to additional information
+or references may also be include if needed. In certain situations where complex
```
Comment (laurenmrice) [inline src/pages/components/form/usage.mdx:288]:
```suggestion
or references may also be included if needed. In certain situations where complex
```

File: src/pages/components/form/usage.mdx [hunk 5]
```diff
@@ -117,215 +115,378 @@ do you gain by collecting this information?
...
+instructions or additional context would be appropriate, tooltips can be used in
+addition to the helper text. However, it is recommended that all pertinent
+assistive information to form completion—such as format content—be presented
+simply and directly as helper text under the field and not be in a tooltip
```
Comment (laurenmrice) [inline src/pages/components/form/usage.mdx:292]:
```suggestion
simply and directly as helper text under the field and not be in a tooltip.
```

File: src/pages/components/form/usage.mdx [hunk 6]
```diff
@@ -117,215 +115,378 @@ do you gain by collecting this information?
...
+![Example of text input with a tooltip](images/form-usage-6.png)
 
 </Column>
 </Row>
 
-## Validation and errors
+#### Tooltips in fluid components
 
-### Error messaging
+Fluid inputs are an exception to the no critial information in tooltips rule.
```
Comment (laurenmrice) [inline src/pages/components/form/usage.mdx:304]:
```suggestion
Fluid inputs are an exception to the no critical information in tooltips rule.
```

File: src/pages/components/form/usage.mdx [hunk 7]
```diff
@@ -117,215 +115,378 @@ do you gain by collecting this information?
...
+Due to their architectural nature, fluid inputs do not provide the allotted
+space under the field needed for helper text. All assistive text wither
```
Comment (laurenmrice) [inline src/pages/components/form/usage.mdx:306]:
```suggestion
space under the field needed for helper text. All assistive text whether
```

File: src/pages/components/form/usage.mdx [hunk 8]
```diff
@@ -117,215 +115,378 @@ do you gain by collecting this information?
...
+pertinent or additional is provided through a tooltip when using fluid inputs.
```
Comment (laurenmrice) [inline src/pages/components/form/usage.mdx:307]:
```suggestion
pertinent or additional, is provided through a tooltip when using fluid inputs.
```

File: src/pages/components/form/usage.mdx [hunk 9]
```diff
@@ -117,215 +115,378 @@ do you gain by collecting this information?
...
-fix it. First, inform the user what has happened, then provide guidance on next
-steps or possible resolutions.
-[Inline notifications](/components/notification/usage) can be used to state the
-general problem with the users input, such as _“Please input the required
-fields.”_ Inline Notifications can occur pre or post submission, depending on
-the type of data the user is inputting.
+<Row>
+<Column colLg={8}>
+
+![Example of a fluid input using a tooltip for help](images/form-usage-7.png)
```
Comment (laurenmrice) [inline src/pages/components/form/usage.mdx:312]: This image has a `etc.` typo as well.
Screen Shot 2023-01-24 at 1 01 41 PM

File: src/pages/components/form/usage.mdx [hunk 10]
```diff
@@ -117,215 +115,378 @@ do you gain by collecting this information?
...
 
+### Form controls
+
+Carbon provides a variety of data input components that enable a user to make a
+enter or select data. The components are divided into three different
+interaction types.
+
+| Type                                                         | Overview                                                                                                                                                                                                            |
+| ------------------------------------------------------------ | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| [Free form inputs](/patterns/forms-pattern/#text-inputs)     | Free-form text inputs are the most commonly used components in forms. They allow a user to enter any configuration of a letters or numbers. Common free form inputs include: text input and text area.              |
```
Comment (laurenmrice) [inline src/pages/components/form/usage.mdx:406]:
```suggestion
| [Free form inputs](/patterns/forms-pattern/#text-inputs)     | Free-form text inputs are the most commonly used components in forms. They allow a user to enter any configuration of letters or numbers. Common free-form inputs include: text input and text area.              |
```

File: src/pages/components/form/usage.mdx [hunk 11]
```diff
@@ -117,215 +115,378 @@ do you gain by collecting this information?
...
+</Tab>
+
+</Tabs>
 
 </Column>
 </Row>
 
+### Form controls
+
+Carbon provides a variety of data input components that enable a user to make a
```
Comment (laurenmrice) [inline src/pages/components/form/usage.mdx:400]:
```suggestion
Carbon provides a variety of data input components that enable a user to
```

File: src/pages/components/form/usage.mdx [hunk 12]
```diff
@@ -117,215 +115,378 @@ do you gain by collecting this information?
...
+
+<Tab label="Fixed">
+
+![Examples of fixed input states](images/form-usage-states-fixed.png)
+
+</Tab>
+
+<Tab label="Fluid">
+
+![Examples fluid input states](images/form-usage-states-fluid.png)
```
Comment (laurenmrice) [inline src/pages/components/form/usage.mdx:439]: The warning helper text should be $text-primary in the fluid image.
Screen Shot 2023-01-24 at 2 31 42 PM

File: src/pages/components/form/usage.mdx [hunk 13]
```diff
@@ -117,215 +115,378 @@ do you gain by collecting this information?
...
 <Row>
-<Column colLg={8}>
+<Column colLg={12}>
 
-![Example of product configuration properties using the required pattern](images/form-usage-required.png)
+<Tabs>
+
+<Tab label="Fixed">
+
+![Examples of fixed input states](images/form-usage-states-fixed.png)
```
Comment (laurenmrice) [inline src/pages/components/form/usage.mdx:433]: The skeleton state input field needs to bottom align with the disabled field to the left of it in the fixed image.
Screen Shot 2023-01-24 at 2 32 48 PM

File: src/pages/components/form/usage.mdx [hunk 14]
```diff
@@ -117,215 +115,378 @@ do you gain by collecting this information?
...
+
+## Additional guidance
+
+The form pattern provides additional and elaborative guidance to the information
+found with the form component. Below is a list of key sections available in the
+pattern.
+
+| Topic                                                                             | Overview                                                                                                        |
+| --------------------------------------------------------------------------------- | --------------------------------------------------------------------------------------------------------------- |
+| [Buttons in forms](/patterns/forms-pattern/#buttons])                             | Provides guidance on button alignment, emphasis, placement and naming in relation to a form.                    |
```
Comment (laurenmrice) [inline src/pages/components/form/usage.mdx:456]:
```suggestion
| [Buttons in forms](/patterns/forms-pattern/#buttons])                             | Provides guidance on button alignment, emphasis, placement, and naming in relation to a form.                    |
```

File: src/pages/components/form/usage.mdx [hunk 15]
```diff
@@ -117,215 +115,378 @@ do you gain by collecting this information?
...
+| [Errors and validation](/forms-pattern/#errors-and-validation)                    | Effective and immediate error messaging can help the user to understand the problem and how to fix it.          |
+| [Form layout](/patterns/forms-pattern/#designing-a-form)                          | Additional guidance around layout, spacing, columns in a form.                                                  |
```
Comment (laurenmrice) [inline src/pages/components/form/usage.mdx:458]:
```suggestion
| [Form layout](/patterns/forms-pattern/#designing-a-form)                          | Additional guidance around layout, spacing, and columns in a form.                                                  |
```

File: src/pages/components/form/style.mdx [hunk 16]
```diff
@@ -47,20 +42,20 @@ for specific typography details on each component.
...
 content and layout of the design. On mobile, forms can only have one column.
 
-### Recommended
+### Recommended for fixed forms
 
 | Element              | Property      | px / rem | Spacing token |
 | -------------------- | ------------- | -------- | ------------- |
 | Form items           | margin-bottom | 32 / 2   | `$spacing-07` |
-| Title area           | margin-bottom | 32 / 2   | `$spacing-07` |
+| Title area           | margin-bottom | 40 / 2   | `$spacing-08` |
```
Comment (laurenmrice) [inline src/pages/components/form/style.mdx:56]:
```suggestion
| Title area           | margin-bottom | 40 / 2.5   | `$spacing-08` |
```

File: src/pages/components/form/style.mdx [hunk 17]
```diff
@@ -72,9 +67,18 @@ content and layout of the design. On mobile, forms can only have one column.
   Structure and spacing measurements for a double column form | px / rem
 </Caption>
 
+### Recommended for fluid forms
+
+| Element              | Property      | px / rem | Spacing token |
+| -------------------- | ------------- | -------- | ------------- |
+| Form items           | margin-bottom | 0 px     | –             |
+| Title area           | margin-bottom | 40 / 2   | `$spacing-08` |
```
Comment (laurenmrice) [inline src/pages/components/form/style.mdx:75]:
```suggestion
| Title area           | margin-bottom | 40 / 2.5   | `$spacing-08` |
```

**Review by laurenmrice — CHANGES_REQUESTED** [review]
Looks really good! ⭐️ Just a couple of typos in content/images.

**Comment by aagonzales** [comment]
Ok updates made, thanks of checking. I forgot to run it through Grammarly 😵‍💫

**Review by laurenmrice — APPROVED** [review]
🚀

**Review by tw15egan — APPROVED** [review]
Looks great 🎉

**Pull request merged**